File: 0048-rotate-image/0048-rotate-image.cs
public class Solution {
    public void Rotate(int[][] matrix) {
        int n = matrix.Length -1;
        for (int i = 0; i < (n + 2) / 2; i ++) {
            for (int j = 0; j < (n +1)/ 2; j++) {
                
                int temp = matrix[n  - j][i];
                matrix[n - j][i] = matrix[n  - i][n - j];
                matrix[n  - i][n - j ] = matrix[j][n  -i];
                matrix[j][n -  i] = matrix[i][j];
                matrix[i][j] = temp;
            }
        }
    }
}

File: 0048-rotate-image/0048-rotate-image.py
class Solution:
    def rotate(self, matrix: List[List[int]]) -> None:
        """
        Do not return anything, modify matrix in-place instead.
        """
        rn = len(matrix)
        n = rn - 1

        for i in range((rn + 1)//2):
            for j in range(rn//2):
                t = matrix[i][j]
                matrix[i][j] = matrix[n-j][i]
                matrix[n-j][i] = matrix[n-i][n-j]
                matrix[n-i][n-j] = matrix[j][n-i]
                matrix[j][n-i] = t